apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/block-editor-nux.js

```diff
@@ -52,7 +52,8 @@ registerPlugin( 'wpcom-block-editor-nux', {
 
 		// Open patterns panel before Welcome Tour if necessary (e.g. when using Blank Canvas theme)
 		// Do this only when Welcome Tour is not manually opened.
-		if ( variant === BLANK_CANVAS_VARIANT && ! isManuallyOpened ) {
+		// NOTE: at the moment, 'starter-page-templates' assets are not loaded on /site-editor/ page so 'setOpenState' may be undefined
+		if ( variant === BLANK_CANVAS_VARIANT && ! isManuallyOpened && setOpenState ) {
```

Comment:
Can't we include this check in the `useEffect` and add `setOpenState` to the dependency array?

Reply:
If you're referring to the [useEffect above](https://github.com/Automattic/wp-calypso/blob/cc1bbf1c46b3d3daecd2e66a95703e82b13abced/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/block-editor-nux.js#L43-L47), we use it to check just the status of the welcome tour but that's independent of the ['automattic/starter-page-layout' store registration](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/store.ts#L25-L35) used to open the Pattern Picker Modal.

Reply:
Yep, I was indeed referring to that `useEffect` method ;) But that makes sense and prevents us from using it as a check for `setOpenState`.